src/server/api/lib/message-sending.js

```diff
@@ -18,16 +18,15 @@ export async function getLastMessage({ contactNumber, service }) {
 
 export async function saveNewIncomingMessage(messageInstance) {
   if (messageInstance.service_id) {
-    const countResult = await r.getCount(
-      r.knex("message").where("service_id", messageInstance.service_id)
-    );
+    const [countResult] = await r
```

Comment:
A theory about the failure was that this was getting a count rather than a single message -- it therefore deadlocked on two of the same messages coming in, because it didn't need to find just one, but get a full count.

Reply:
This makes sense, though it would be good to rename this variable, since it is no longer the count (unless I'm misunderstanding the change).